### PR TITLE
Jbaublitz issue project 310

### PR DESCRIPTION
**Pull request by mulkieran**
Related https://github.com/stratis-storage/project/issues/310

### CODE REVIEW

**Review by drckeefe — APPROVED** [review]

**Comment by mulkieran** [comment]
rebased.

**Comment by mulkieran** [comment]
rebased.

**Comment by mulkieran** [comment]
Why not use the three argument ```handle_action``` macro for all the calls. Is it more expensive?

File: src/dbus_api/consts.rs [hunk 1]
```diff
@@ -21,6 +21,7 @@ pub const POOL_NAME_PROP: &str = "Name";
 pub const POOL_UUID_PROP: &str = "Uuid";
 pub const POOL_HAS_CACHE_PROP: &str = "HasCache";
 pub const POOL_ENCRYPTED_PROP: &str = "Encrypted";
+pub const POOL_STATE_PROP: &str = "State";
```
Comment (mulkieran) [inline src/dbus_api/consts.rs:24]: Would like to call this something less generic than "State".

Comment (jbaublitz) [reply src/dbus_api/consts.rs:24]: Sounds good! I'll go with `AvailableActions` as we decided in stand up.

**Comment by jbaublitz** [comment]
> Why not use the three argument `handle_action` macro for all the calls. Is it more expensive? Most of the time the D-Bus layer won't have to analyze the error, so I feel it ought not to be. Minor code bloat.

This seems reasonable to me as a future proofing measure!

**Comment by mulkieran** [comment]
stratis-cli test should fail

**Pull request closed without merging**